pkg/volume/gcepd/attacher.go

```diff
@@ -141,6 +141,45 @@ func (attacher *gcePersistentDiskAttacher) VolumesAreAttached(specs []*volume.Sp
 	return volumesAttachedCheck, nil
 }
 
+func (attacher *gcePersistentDiskAttacher) BulkVerifyVolumes(volumesByNode map[types.NodeName][]*volume.Spec) (map[types.NodeName]map[*volume.Spec]bool, error) {
+	volumesAttachedCheck := make(map[types.NodeName]map[*volume.Spec]bool)
+	deviceNamesByNode := make(map[types.NodeName][]string)
+
+	for nodeName, volumeSpecs := range volumesByNode {
+		deviceNames := []string{}
+		for _, spec := range volumeSpecs {
+			volumeSource, _, err := getVolumeSource(spec)
+			if err != nil {
+				klog.Errorf("Error getting volume (%q) source : %v", spec.Name(), err)
```

Comment:
add some more contextual info here like "while bulkverifyvolumes-ing" or something. Also why is this error not bad enough that we would stop processing and instead ignore it

Reply:
I guess with a big cluster, you can for some reason have a malformed volume spec but it shouldn't prevent all the other ok ones from being checked. I can see how that can be an issue though, since if an error is not returned the malformed spec is never dealt with, but I think that is better than not checking any at all.

Reply:
I'm unsure if this is a scenario that is possible for users to get into (and pass API validation).

Reply:
Do you know who originally wrote this for VolumesAreAttached? Maybe they can have a better explanation for why the log is included or if there is a scenario we're not thinking of.

Reply:
@jingxu97

Reply:
The volume verification mechanism exists as a fail-safe to correct volumes that would have been incorrectly detached from nodes. So, it is *only* designed to cause attaches and never detaches. 

And since it verifies entire cluster it runs less frequently than other volume operations. So, if we come across a volume which we can't verify (like `getVolumeSource` failed!) then it is best to skip it rather than letting the whole operation fail.

Reply:
@hantaowang I think my thought about it is same as yours. I think if our API validation does its job correctly, the error should never happen. Also, this function is used by other functions such as Attach(), if it returns error, attach will fail and verify function will not be called at all. Unless someone somehow modify the PV and remove its PD name after volume is attached, which would be really strange.